rotors_gazebo_plugins/src/gazebo_wind_plugin.cpp

```diff
@@ -54,38 +57,105 @@ void GazeboWindPlugin::Load(physics::ModelPtr _model, sdf::ElementPtr _sdf) {
-
-  wind_direction_.Normalize();
-  wind_gust_direction_.Normalize();
-  wind_gust_start_ = common::Time(wind_gust_start);
-  wind_gust_end_ = common::Time(wind_gust_start + wind_gust_duration);
-
-  wind_force_n_ = NormalDistribution(0, sqrt(wind_force_variance_));
-  wind_gust_force_n_ = NormalDistribution(0, sqrt(wind_gust_force_variance_));
-  wind_speed_n_ = NormalDistribution(0, sqrt(wind_speed_variance_));
+
+  // Should a custom static wind field be used? Parameter to be set in techpod_base.xacro
+  getSdfParam<bool>(_sdf, "customStaticWindField", custom_static_wind_field_, custom_static_wind_field_);
+
+  if (!custom_static_wind_field_)
+  {
+    // Get the wind params from SDF.
+    getSdfParam<double>(_sdf, "windForceMean", wind_force_mean_, wind_force_mean_);
+    getSdfParam<double>(_sdf, "windForceVariance", wind_force_variance_, wind_force_variance_);
+    getSdfParam<math::Vector3>(_sdf, "windDirection", wind_direction_, wind_direction_);
+    // Get the wind speed params from SDF
+    getSdfParam<double>(_sdf, "windSpeedMean", wind_speed_mean_, wind_speed_mean_);
+    getSdfParam<double>(_sdf, "windGustStart", wind_gust_start, wind_gust_start);
+    getSdfParam<double>(_sdf, "windGustDuration", wind_gust_duration, wind_gust_duration);
+    getSdfParam<double>(_sdf, "windGustForceMean", wind_gust_force_mean_, wind_gust_force_mean_);
+    getSdfParam<double>(_sdf, "windGustForceVariance", wind_gust_force_variance_, wind_gust_force_variance_);
+    getSdfParam<math::Vector3>(_sdf, "windGustDirection", wind_gust_direction_, wind_gust_direction_);
+
+    wind_direction_.Normalize();
+    wind_gust_direction_.Normalize();
+    wind_gust_start_ = common::Time(wind_gust_start);
+    wind_gust_end_ = common::Time(wind_gust_start + wind_gust_duration);
+
+    wind_force_n_ = NormalDistribution(0, sqrt(wind_force_variance_));
+    wind_gust_force_n_ = NormalDistribution(0, sqrt(wind_gust_force_variance_));
+    wind_speed_n_ = NormalDistribution(0, sqrt(wind_speed_variance_));
+  }
+  else
+  {
+    getSdfParam<math::Vector3>(_sdf, "windDirection", wind_direction_, wind_direction_);
+    getSdfParam<double>(_sdf, "windSpeedMean", wind_speed_mean_, wind_speed_mean_);
+    getSdfParam<std::string>(_sdf, "windPath", wind_path_, wind_path_);
+    // Read the txt file containing the wind field data, save it to a 2D array of math::Vector3 elements
+    std::ifstream fin;
+    fin.open(wind_path_);
+    if (fin.is_open())
+    {
+      math::Vector3 position;
+      math::Vector3 wind;
+      double value = 0;
+      bool fill = true;
+
+      while (fin >> value)
+      {
+        position.x = value;
+        fin >> position.y;
+        fin >> position.z;
+        fin >> wind.x;
+        fin >> wind.y;
+        fin >> wind.z;
+        wind_field_[0].push_back(position);
+        wind_field_[1].push_back(wind);
+
+        // Fill the x and y coordinate vectors
+        for (int i = 0; i < coords_x_.size(); i++)
```

Comment:
The indentation was correct, but the readability not optimal. I inserted those braces and more comments to improve it.